src/main/java/com/snowflake/kafka/connector/records/SnowflakeRecordContent.java

```diff
@@ -30,19 +52,161 @@
+      this.brokenData = data.toString().getBytes();
+      return;
+    }
+    this.isBroken = false;
     this.brokenData = null;
   }
 
+  /**
+   * Convert this object, in the org.apache.kafka.connect.data format, into a JSON object, returning both the schema
+   * and the converted object.
+   * @param schema schema of the object
+   * @param logicalValue object to be converted
+   * @return a JsonNode of the object
+   */
+  private static JsonNode convertToJson(Schema schema, Object logicalValue) {
+    if (logicalValue == null) {
+      if (schema == null) // Any schema is valid and we don't have a default, so treat this as an optional schema
+        return null;
+      if (schema.defaultValue() != null)
+        return convertToJson(schema, schema.defaultValue());
+      if (schema.isOptional())
+        return JsonNodeFactory.instance.nullNode();
+      throw new DataException("Conversion error: null value for field that is required and has no default value");
+    }
+
+    Object value = logicalValue;
+    try {
+      final Schema.Type schemaType;
+      if (schema == null) {
+        schemaType = ConnectSchema.schemaType(value.getClass());
+        if (schemaType == null)
+          throw new DataException("Java class " + value.getClass() + " does not have corresponding schema type.");
+      } else {
+        schemaType = schema.type();
+      }
+      switch (schemaType) {
+        case INT8:
+          return JsonNodeFactory.instance.numberNode((Byte) value);
+        case INT16:
+          return JsonNodeFactory.instance.numberNode((Short) value);
+        case INT32:
+          if (schema != null && Date.LOGICAL_NAME.equals(schema.name())) {
+            return JsonNodeFactory.instance.textNode(ISO_DATE_FORMAT.format((java.util.Date) value));
+          }
+          if (schema != null && Time.LOGICAL_NAME.equals(schema.name())) {
+            return JsonNodeFactory.instance.textNode(TIME_FORMAT.format((java.util.Date) value));
+          }
+          return JsonNodeFactory.instance.numberNode((Integer) value);
+        case INT64:
+          String schemaName = schema.name();
+          if(Timestamp.LOGICAL_NAME.equals(schemaName)){
+            return JsonNodeFactory.instance.numberNode(Timestamp.fromLogical(schema, (java.util.Date) value));
+          }
+          return JsonNodeFactory.instance.numberNode((Long) value);
+        case FLOAT32:
+          return JsonNodeFactory.instance.numberNode((Float) value);
+        case FLOAT64:
+          return JsonNodeFactory.instance.numberNode((Double) value);
+        case BOOLEAN:
+          return JsonNodeFactory.instance.booleanNode((Boolean) value);
+        case STRING:
+          CharSequence charSeq = (CharSequence) value;
+          return JsonNodeFactory.instance.textNode(charSeq.toString());
+        case BYTES:
+          if (Decimal.LOGICAL_NAME.equals(schema.name())) {
+            return JsonNodeFactory.instance.numberNode((BigDecimal) value);
```

Comment:
Very late to the party here, but my team at work is having trouble with avro logical decimal types being stored in snowflake properly (we're seeing what's reported in https://github.com/snowflakedb/snowflake-kafka-connector/issues/122 ) and this looks odd to me. 

Two things look odd:
1. this does a cast from a byte array but does not specify the precision or scale of the logical data type to the resulting BigDecimal
2. json doesn't support arbitrary precision in its numeric type (it's basically a `double`). The only safe way to represent a java BigDecimal in json is to use a string